src/context/wishlist-context.js

```diff
@@ -0,0 +1,41 @@
+import {createContext,useReducer, useContext} from 'react';
+
+const wishListContext = createContext(null);
+
+const WishListContextProvider = ({children}) => {
+  const reducerFunction = (wishListState, {type,payload}) => {
```

Comment:
you can write this reducer function outside of ContextProvider. It will be better for readability. If you can then create a folder called reducers and keep all the reducer functions in it